### ISSUE TITLE
How to obtain better precision

### ISSUE THREAD

**Issue by AiMizuno** [bryanyzhu/two-stream-pytorch#32]
I set the parameter as readme but obtain 81.76% for spatial stream on split 1 of UCF101 dataset using ResNet152. How to achieve better performance? Thanks.

**Comment by bryanyzhu** [comment]
Sorry for the slow reply, did you use opencv to decode the video to frames? Since many people have reproduced the number, I doubt the lower performance is because of your input images.

**Comment by AiMizuno** [comment]
Thanks. The input actually affect the performance. And I notice the precision during validation is lower than testing. Is it also related to the input?

**Comment by bryanyzhu** [comment]
I don't know if I understand you correctly. In current code base, **the validation set is the testing set**. 

The reason validation accuracy is lower is because: for validation, we only randomly pick 1 frame from each video and do the prediction. But for testing, we pick 25 frames and do 10-crop validation on each frame to get the final results. Simply put, we only do 1 forward pass to get prediction during validation, but we do 250 forward passes to get prediction during testing. The average of 250 results is definitely better and more robust than only 1 result. 

But if this is not what you mean by validation and testing, just let me know. We can discuss further.

**Comment by AiMizuno** [comment]
That is what I mean. I am clear now. Thank you so much.
